### DISCUSSION TITLE
Multi-map / Reduce indexes with IndexDefinitionBuilder

### DISCUSSION THREAD

**Discussion by toochevere** [ravendb/ravendb#15464] [Q&A] [answered]
I am using `IndexDefinitionBuilder` (C# client) to create indexes.  The indexes are being created in a library and the expressions that form the basis of the indexes will vary according to the use of the library.  

Up until now these have only been map or single map / reduce indexes.  Recently a use case came up where a multi-map / reduce index would be a very clean solution to the particular problem.  However `IndexDefinitionBuilder` only takes one map expression. 

How can build a multi-map  reduce index with `IndexDefinitionBuilder` or something similar?   A workaround I can see is to dynamically compile a class that uses [AbstractMultiMapIndexCreationTask](https://ravendb.net/docs/article-page/5.3/csharp/indexes/multi-map-indexes) which feels very heavy-handed, but it would let me stay in pure C#.  Any better ideas would be welcome.

**Comment by garayx** [comment, marked as answer]
Hi

Yes, using the `IndexDefinitionBuilder` is supported for single map or map-reduce index for now. Using [AbstractMultiMapIndexCreationTask](https://ravendb.net/docs/article-page/5.3/csharp/indexes/multi-map-indexes) for creation of multi map index is the correct way.
